src/modules/Elsa.Retention/Jobs/CleanupJob.cs

```diff
@@ -0,0 +1,66 @@
+using Elsa.Common.Contracts;
+using Elsa.Retention.Extensions;
+using Elsa.Retention.Options;
+using Elsa.Workflows.Management;
+using Elsa.Workflows.Management.Entities;
+using Elsa.Workflows.Management.Enums;
+using Elsa.Workflows.Management.Filters;
+using Elsa.Workflows.Management.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Elsa.Retention.Jobs;
+
+  /// <summary>
+    /// Deletes all workflow instances that are older than a specified threshold (configured through options).
+    /// </summary>
+    public class CleanupJob
+    {
+        private readonly IWorkflowInstanceManager _workflowInstanceManager;
+        private readonly CleanupOptions _options;
+        private readonly ILogger _logger;
+        private readonly ISystemClock _systemClock;
+
+        /// <summary>
+        /// Creates a new cleanup job
+        /// </summary>
+        /// <param name="workflowInstanceManager"></param>
+        /// <param name="options"></param>
+        /// <param name="systemClock"></param>
+        /// <param name="logger"></param>
+        public CleanupJob(
+            IWorkflowInstanceManager workflowInstanceManager,
+            IOptions<CleanupOptions> options,
+            ISystemClock systemClock,
+            ILogger<CleanupJob> logger)
+        {
+            _systemClock = systemClock;
+            _options = options.Value;
+            _logger = logger;
+            _workflowInstanceManager = workflowInstanceManager;
+        }
+
+        /// <summary>
+        /// Executes the cleanup job
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        public async Task ExecuteAsync(CancellationToken cancellationToken = default)
+        {
+            var threshold = _systemClock.UtcNow.Subtract(_options.TimeToLive);
+            WorkflowInstanceFilter specification = _options.WorkflowInstanceFilter.Clone();
```

Comment:
I would create a specific filter type for this (Something like `WorkflowInstanceRetentionFilter`. This should have the opportunity to filter (including or excluding) based on a collection of DefinitionIds or DefinitionVersionIds. This would prevent filtering on properties that do not make much sence to be used for retention (like the InstanceId).
This filter can then be mapped to the WorkflowInstanceFilter so that we can still reuse the current BulkDeleteAsync function.